pkg/image/sif.go

```diff
@@ -127,14 +152,16 @@ func (f *sifFormat) initializer(img *Image, fileinfo os.FileInfo) error {
 				Size:   uint64(desc.Filelen),
 				Name:   desc.GetName(),
 			}
-			switch fstype {
-			case sif.FsSquash:
-				partition.Type = SQUASHFS
-			case sif.FsExt3:
-				partition.Type = EXT3
-			default:
-				partition.Type = uint32(fstype)
+
+			htype, err := checkPartitionType(img, fstype, desc.Fileoff)
+			if err != nil && err != ErrUnknownFsType {
+				return fmt.Errorf("while checking data partition header: %s", err)
+			} else if err == ErrUnknownFsType {
+				img.Partitions[0].Type = uint32(fstype)
```

Comment:
I know this was here before but it strikes me as sketchy. Doesn't this assume the constants of https://github.com/sylabs/singularity/blob/c5e243fc45fc5b55f315f7e39255cb95201d4f17/pkg/image/image.go#L21 and https://github.com/sylabs/sif/blob/640d2f2a0ef5219f0066d277e465b6c18a046fc8/pkg/sif/sif.go#L147 are kept in sync? Is this ensured at another place?

Reply:
We can't really sync those constants with the SIF ones because the image code is generic and must handle different format like sandbox, ext3 only, squashfs only, SIF (which can contain ext3/squashfs image), so it doesn't hurt to have different values there.

Reply:
But this code basically says: We have an unknown file system type, just assume that the incoming type id (type is `Fstype` from SIF package) is a valid Type here (like `SQUASHFS`) even though we may not even know it.

This works for e.g. `fstype==FsEncryptedSquashfs` because it happens to hold `uint32(FsEncryptedSquashfs) == ENCRYPTSQUASHFS`, but what if the enums(?) go out of sync?

Reply:
I initially thought to keep the entry for a possible use by the caller as it's a public package but as it's only for overlay partition, we could probably just ignore those unknow fstype entries to avoid confusion, thoughts ?

Reply:
Not sure if I understand correctly but IMO an unknown fs type is an error and should be reported as such. So I'd remove lines 159&160 and the check for `err != ErrUnknownFsType ` from 157